Jenkinsfile.trigger

```diff
@@ -161,12 +181,47 @@ for (buildObj in queue) {
 				quietPeriod: 5, // seconds
 			)
 			// TODO do something useful with "res.result" (especially "res.result != 'SUCCESS'")
-			// (maybe store "res.startTimeInMillis + res.duration" as endTime so we can implement some amount of backoff somehow?)
 			echo(res.result)
 			if (res.result != 'SUCCESS') {
+				def c = 1
+				if (pastFailedJobs[buildObj.buildId]) {
+					c += pastFailedJobs[buildObj.buildId].count
+				}
+				// TODO maybe implement some amount of backoff? keep first url/endtime?
+				newFailedJobs[buildObj.buildId] = [
+					count: c,
+					identifier: identifier,
+					url: res.absoluteUrl,
+					endtime: (res.startTimeInMillis + res.duration) / 1000.0, // convert to seconds
+				]
+
 				// "catchError" is the only way to set "stageResult" :(
 				catchError(message: 'Build of "' + identifier + '" failed', buildResult: 'UNSTABLE', stageResult: 'FAILURE') { error() }
 			}
 		}
 	}
 }
+
+// save newFailedJobs so we can use it next run as pastFailedJobs
+node {
+	def newFailedJobsJson = writeJSON(json: newFailedJobs, returnText: true)
+	withEnv([
+		'newFailedJobsJson=' + newFailedJobsJson,
+	]) {
+		stage('Archive') {
+			dir('builds') {
+				deleteDir()
+				sh '''#!/usr/bin/env bash
+					set -Eeuo pipefail -x
+
+					jq '.' <<<"$newFailedJobsJson" | tee pastFailedJobs.json
```

Comment:
Should the string `pastFailedJobs.json` be abstracted into a string since it is used several places?

Reply:
Yeah, that could help cut down on typos, but it does get a little annoying to inject properly (we'd probably need to do something like `env.PAST_FAILED_JOBS_FILENAME = ...` or something so that it can be accessed from all the shell invocations without dealing with quoting hell).